STYLE_GUIDE.md

```diff
@@ -0,0 +1,158 @@
+
+---
+
+## 3. Contribution Workflow
+
+* **Branching Strategy:**
+    * **Main Branch:** `main` is the stable production branch.
+    * **Feature Branches:** Create branches from `main` for all changes.
+* **Pull Requests (PRs):**
+    * **Atomic Changes:** PRs should be small enough to be reviewed in one sitting. Large, sprawling PRs may be requested to be split.
+    * **Review Requirement:** All PRs require review from at least two Maintainers.
+    * **CI/CD Gates:**
+      * **Standard:** All PRs must generally pass automated checks (linting, testing, build) before merging.
+      * **Exceptions:** We recognize that checks may occasionally fail due to external issues outside our control or transient flakes that pass locally. In these rare instances, **maintainers can discuss and agree on exceptions** to merge specific PRs despite a red status.
+    * **Pull Request Title Format**
+      * **Format:** `<type>: <description>`
+      * **Example:** `feat: implement oscal validation logic`
+
+* **Commit Messages:**
+    * Follow the **Conventional Commits** [specification](https://www.conventionalcommits.org/).
+
+---
+
+## 4. Centrally manages configuration files, templates, and standardized CI/CD workflows
+
+We should centralize workflows, configurations, and templates as much as possible. Refer to [org-infra](https://github.com/complytime/org-infra).
+
+## 5. Coding Standards
+
+### Guidelines for all programming language
+
+- **Empty Line at End of File**: Ensure that all files include an empty line at the end. This helps with version control diffs and adheres to POSIX standards.
+- The pre-commit and pre-push hooks can be configured by installing [pre-commit](https://pre-commit.com/) and running code repo specific `make` commands.
+- **Testing**: Write tests for your code. Use descriptive names for test functions and include edge cases.
+
+### Go (e.g., `complyctl`)
+
+#### General Guidelines
+
+- **File Naming**: Use lowercase letters and underscores for file names (e.g., `my_file.go`).
+- **Package Names**: Use short, concise, and lowercase names for packages. Avoid underscores and mixed caps.
+- **Error Handling**: Always check for errors and handle them appropriately. Return errors to the caller when necessary.
+- **Line Length**: Limit lines to 99 characters when reasonable to improve readability.
+
+#### Licensing and File Headers
+
+```go
+// SPDX-License-Identifier: Apache-2.0
+```
+
+#### Code Formatting
+
+- **Imports**: Use [`goimports`](https://pkg.go.dev/golang.org/x/tools/cmd/goimports) to format and organize imports. Separate standard library, third-party, and project-specific imports with blank lines. Also ensure only necessary imports.
+- **Indentation and Spacing**: Use the [`go fmt`](https://go.dev/blog/gofmt) tool to automatically format your code. This ensures consistent indentation with tabs and proper spacing after commas, colons, and semicolons.
```

Comment:
Thinking if wouldn't be enough to say that formatting should be aligned with native go format tools, (`go fmt`)  instead of being more specific about some rules. WDYT?